octopusdeploy/schema_variable.go

```diff
@@ -55,19 +55,107 @@ func expandVariable(d *schema.ResourceData) *variables.Variable {
 }
 
 func getVariableDataSchema() map[string]*schema.Schema {
-	dataSchema := getVariableSchema()
-	setDataSchema(&dataSchema)
-
 	return map[string]*schema.Schema{
-		"id":       getDataSchemaID(),
-		"space_id": getQuerySpaceID(),
-		"ids":      getQueryIDs(),
-		"variables": {
+		"id": {
 			Computed:    true,
-			Description: "A list of variables that match the filter(s).",
-			Elem:        &schema.Resource{Schema: dataSchema},
-			Optional:    true,
+			Description: "The identifier of the variable to find.",
+			Type:        schema.TypeString,
+		},
+		"name": {
+			Required:    true,
+			Description: "The name of variable to find.",
+			Type:        schema.TypeString,
+		},
+		"owner_id": {
+			Required:    true,
+			Description: "Owner ID for the variable to find.",
+			Type:        schema.TypeString,
+		},
+		"scope": {
+			Description: "As variable names can appear more than once under different scopes, a VariableScope must also be provided",
+			Elem:        &schema.Resource{Schema: getVariableScopeSchema()},
+			Required:    true,
 			Type:        schema.TypeList,
+			MaxItems:    1,
+		},
+		"space_id": getQuerySpaceID(),
```

Comment:
does this mean the user cannot supply the space_id?

Reply:
nvm, it seems that it was the same as before

Reply:
this means they can use it :)